lib/error/error.cpp

```diff
@@ -0,0 +1,110 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Pionix GmbH and Contributors to EVerest
+#include <utils/error/error.hpp>
+
+#include <boost/uuid/random_generator.hpp>
+#include <boost/uuid/uuid.hpp>
+#include <boost/uuid/uuid_io.hpp>
+
+#include <utils/error/error_exceptions.hpp>
+
+namespace Everest {
+namespace error {
+
+UUID::UUID() {
+    uuid = boost::uuids::to_string(boost::uuids::random_generator()());
+}
+
+UUID::UUID(const std::string& uuid_) : uuid(uuid_) {
+}
+
+bool UUID::operator<(const UUID& other) const {
+    return uuid < other.uuid;
+}
+
+bool UUID::operator==(const UUID& other) const {
+    return this->uuid == other.uuid;
+}
+
+bool UUID::operator!=(const UUID& other) const {
+    return !(*this == other);
+}
+
+std::string UUID::to_string() const {
+    return uuid;
+}
+
+Error::Error() :
+    Error("", "", "", ImplementationIdentifier("", ""), Severity::Low, date::utc_clock::now(), UUID(), State::Active) {
```

Comment:
Is this really a useful default? Looks like not much insight can be gained from an error like this

Reply:
I needed it somewhere in the development process. Can be useful to have the oportunity to build up an error object starting with an "empty" one. But I would remove it for now since it is no where used at the moment